Use "/" in $routes->resource and $routes->presenter to cause access failure.

vocaloidat opened the pull request:
In the official document: https://codeigniter.com/user_guide/incoming/restful.html
Tutorial:
$routes->resource('api/photo');
$routes->presenter('admin/photos');
Wrong execution guidance.
#### Here, in the official statement, https://codeigniter.com/user_guide/incoming/restful.html

**Sorry, my English is not clear.**

can be seen:

```php
$routes->resource('api/photo');
$routes->presenter('admin/photos');
```

The code generated above is mapped (incorrect):

```php
$routes->put('api/photo/(:segment)','Api/Photo::update/$1');
....
$routes->post('admin/photos/(:segment)', 'Admin/Photos::update/$1');
```

But the standard should be (correct)

```php
$routes->put('api/photo/(:segment)','Api\Photo::update/$1');
....
$routes->post('admin/photos/(:segment)', 'Admin\Photos::update/$1');
```

Yes=‘\’, not=‘/’.
The command space USES = ‘\’, not=  ‘/’ as in the URL.

### So I added this code to prevent errors.



According to the current version, this sub-operation is wrong, and the URL link address (put, delete, etc.) cannot be accessed according to this. This submission request is required to complete the official complete update of the framework.

##### So I added this code to prevent errors.

MGatner commented:
I think these are supposed to be routes, not namespaces. I will have to look close when not on mobile.

vocaloidat commented:
> 我认为这些应该是路由，而不是名称空间。不使用手机时，我将不得不靠近看。

Yes, but it was executed according to the source code, and an error occurred. This modification can avoid mistakes.

vocaloidat commented:
> I think these are supposed to be routes, not namespaces. I will have to look close when not on mobile.

Yes, but it was executed according to the source code, and an error occurred. This modification can avoid mistakes.

vocaloidat commented:
> I think these are supposed to be routes, not namespaces. I will have to look close when not on mobile.

Here, in the official statement, https://codeigniter.com/user_guide/incoming/restful.html
can be seen:
$routes->resource('api/photo');
$routes->presenter('admin/photos');
According to the current version, this operation is wrong, and this submission request is required to complete the official complete update of the framework.

vocaloidat commented:
> I think these are supposed to be routes, not namespaces. I will have to look close when not on mobile.

Here, in the official statement, https://codeigniter.com/user_guide/incoming/restful.html
can be seen:
$routes->resource('api/photo');
$routes->presenter('admin/photos');
According to the current version, this sub-operation is wrong, and the URL link address (put, delete, etc.) cannot be accessed according to this. This submission request is required to complete the official complete update of the framework.

vocaloidat commented:
> I think these are supposed to be routes, not namespaces. I will have to look close when not on mobile.

#### Here, in the official statement, https://codeigniter.com/user_guide/incoming/restful.html

can be seen:

```php
$routes->resource('api/photo');
$routes->presenter('admin/photos');
```

The code generated above is mapped (incorrect):

```php
$routes->put('api/photo/(:segment)','Api/Photo::update/$1');
....
$routes->post('admin/photos/(:segment)', 'Admin/Photos::update/$1');
```

But the standard should be (correct)

```php
$routes->put('api/photo/(:segment)','Api\Photo::update/$1');
....
$routes->post('admin/photos/(:segment)', 'Admin\Photos::update/$1');
```

Yes=‘\’, not=‘/’.

### So I added this code to prevent errors.



According to the current version, this sub-operation is wrong, and the URL link address (put, delete, etc.) cannot be accessed according to this. This submission request is required to complete the official complete update of the framework.

##### So I added this code to prevent errors.

vocaloidat commented:
> I think these are supposed to be routes, not namespaces. I will have to look close when not on mobile.

#### Here, in the official statement, https://codeigniter.com/user_guide/incoming/restful.html
@MGatner 
can be seen:

```php
$routes->resource('api/photo');
$routes->presenter('admin/photos');
```

The code generated above is mapped (incorrect):

```php
$routes->put('api/photo/(:segment)','Api/Photo::update/$1');
....
$routes->post('admin/photos/(:segment)', 'Admin/Photos::update/$1');
```

But the standard should be (correct)

```php
$routes->put('api/photo/(:segment)','Api\Photo::update/$1');
....
$routes->post('admin/photos/(:segment)', 'Admin\Photos::update/$1');
```

Yes=‘\’, not=‘/’.
The command space USES = ‘\’, not= ‘/’ as in the URL.
### So I added this code to prevent errors.



According to the current version, this sub-operation is wrong, and the URL link address (put, delete, etc.) cannot be accessed according to this. This submission request is required to complete the official complete update of the framework.

##### So I added this code to prevent errors.

michalsn commented on system/Router/RouteCollection.php:
```diff
@@ -807,7 +807,8 @@ public function resource(string $name, array $options = null): RouteCollectionIn
 		// resources are sent to, we need to have a new name
 		// to store the values in.
 		$new_name = ucfirst($name);
-
+		// replace / as \
+        $new_name = str_replace("/","\\",$new_name);
```
Can we change it to one-liner which will `ucfirst` all parts of the namespace?
```php
$new_name = implode("\\", array_map('ucfirst', explode('/', $name)));
```

vocaloidat replied:
Nice, I'm sorry that due to regional reasons (China), there was a problem with VPN, which was not timely.

michalsn commented on system/Router/RouteCollection.php:
```diff
@@ -921,7 +922,8 @@ public function presenter(string $name, array $options = null): RouteCollectionI
 		$newName = ucfirst($name);
-
+		// replace / as \
+        $newName = str_replace("/","\\",$newName);
```
Same as above
```php
$newName = implode("\\", array_map('ucfirst', explode('/', $name)));
```

michalsn commented on system/Router/RouteCollection.php:
```diff
@@ -806,8 +806,9 @@ public function resource(string $name, array $options = null): RouteCollectionIn
 		// to store the values in.
-		$new_name = ucfirst($name);
-
+		//$new_name = ucfirst($name);
+		// Michal thought, modifying a single line in all parts of the namespace
```
Please remove these two lines. Changes are tracked by git history so commenting the code is not necessary. Also, mentioning my name in the comment is not worth it :)

michalsn commented on system/Router/RouteCollection.php:
```diff
@@ -920,8 +921,9 @@ public function presenter(string $name, array $options = null): RouteCollectionI
 		// to store the values in.
-		$newName = ucfirst($name);
-
+		//$newName = ucfirst($name);
+		// Michal thought, modifying a single line in all parts of the namespace
```
Same as above :)

michalsn commented:
Thank you for your first contribution to this project!
